docs/DesignDocs/ExtendedEnclaveInitializationData.md

```diff
@@ -0,0 +1,168 @@
+
+An important instance of this problem is Azure Confidential Containers: the base enclave image contains the SGX-LKL runtime, which executes a user container. The container contains arbitrary code, including the `EREPORT` instruction to obtain valid evidence for arbitrary `report_data` – potentially impersonating other containers.
+
+One way to solve this issue is to re-compile and re-sign a new enclave image for every container we launch (with the container data being measured together with the LKL base image). However, this approach is cumbersome as we would like to be able to launch arbitrary containers on demand without having to build and sign a new enclave image every time.
+
+Another class of problems we address is dynamic attestation of the memory and threading configuration. In current OE attestation, these settings (`NumStackPage`, `NumHeapPages`, `NumTCS`) are fixed at enclave signinig time. This means that it is not possible to deploy the same enclave image with different memory and threading configurations.
+
+User Experience
+---------------
+
+The user decides to enable the use of EEID when the enclave base image is signed: while not strictly necessary, we require that EEID enclaves use `NumStackPages=0`, `NumHeapPages=0`, and `NumTCS=0`. This guarantees that enclave images meant to be used with EEID cannot be accidentally initialized with traditional attestation (the enclave would fail to load).
+
+To launch an EEID enclave, the user uses the new `oe_create_enclave_eeid` API (which adds `eeid` to the argument of `oe_create_enclave`):
+```C
+oe_result_t oe_create_enclave_eeid(
+    const char* enclave_path,
+    oe_enclave_type_t enclave_type,
+    uint32_t flags,
+    const oe_enclave_setting_t* settings,
+    uint32_t setting_count,
+    const oe_ocall_func_t* ocall_table,
+    uint32_t ocall_count,
+    oe_eeid_t* eeid,
+    oe_enclave_t** enclave_out)
+```
+
+The additional argument is of type `oe_eeid_t`, which contains the following information:
+
+```C
+typedef struct oe_eeid_t_
+{
+    uint32_t hash_state[10]; /* internal state of the hash at the end of the enclave base image */
+    uint8_t sigstruct[1808]; /* complete sigstruct (or similar object for other TEE types) computed for the base image */
```

Comment:
That's the size of the Intel sigstruct, so that number is from the SGX spec. To support other TEEs with the same structure, this would have to be a variable size buffer. I'll change that.